controller/jobs.go

```diff
@@ -153,6 +154,42 @@ func (c clusterClientWrapper) Hosts() ([]utils.HostClient, error) {
 	return res, nil
 }
 
+func (c clusterClientWrapper) StreamHosts(ch chan utils.HostClient) (stream.Stream, error) {
+	hostChan := make(chan *cluster.Host)
+	stream, err := c.Client.StreamHosts(hostChan)
+	if err == nil {
+		go func() {
+			for {
+				select {
+				case h, ok := <-hostChan:
+					if !ok {
+						return
+					}
+					ch <- h
+				}
+			}
+		}()
+	}
+	return &clusterStream{
+		ch:           ch,
+		parentStream: stream,
+	}, err
+}
+
+type clusterStream struct {
+	ch           chan utils.HostClient
+	parentStream stream.Stream
+}
+
+func (cs *clusterStream) Close() error {
+	close(cs.ch)
```

Comment:
Defer the close of this channel in the goroutine which is writing to it so that we don't cause a panic writing to a closed channel.